qubesmanager/settings.py

```diff
@@ -150,6 +225,25 @@ def __init__(self, vm, qapp, init_page="basic", parent=None):
             self.refresh_apps_button.clicked.connect(
                 self.refresh_apps_button_pressed)
 
+    def clear_threads(self):
+        for thread in self.threads_list:
+            if thread.isFinished():
+                if self.progress:
+                    self.progress.hide()
+                    self.progress = None
+
+                if thread.msg:
+                    (title, msg) = thread.msg
+                    QtGui.QMessageBox.warning(
+                        None,
+                        self.tr(title),
+                        self.tr(msg))
+
+                self.threads_list.remove(thread)
```

Comment:
Modifying the list while iterating over it is a bad idea. Iterate over its copy.

Reply:
Theorically we should have one call to 'clear_threads' for each finished thread so I added a return after removing and added a exception raise if something goes wrong.